Hugo server crashes when lots of content files change within a short period of time

I programatically delete and regenerate directories and markdown files. Running Hugo server at the same time often causes Hugo v0.56.3/extended windows/amd64 to crash. Here ist the log:

```
Change detected, rebuilding site.
2019-08-10 15:41:46.114 +0200
Source changed "C:\\Hugo\\example.com\\content\\en\\cities\\44\\index.md": CREATE
Source changed "C:\\Hugo\\example.com\\content\\en\\cities\\47\\index.md": CREATE
Source changed "C:\\Hugo\\example.com\\content\\en\\cities\\55\\index.md": CREATE
Source changed "C:\\Hugo\\example.com\\content\\en\\cities\\56\\index.md": CREATE
Source changed "C:\\Hugo\\example.com\\content\\en\\cities\\96\\index.md": CREATE
Source changed "C:\\Hugo\\example.com\\content\\en\\cities\\108\\index.md": CREATE
Source changed "C:\\Hugo\\example.com\\content\\en\\cities\\122\\index.md": CREATE
Source changed "C:\\Hugo\\example.com\\content\\en\\cities\\129\\index.md": CREATE
Source changed "C:\\Hugo\\example.com\\content\\en\\cities\\147\\index.md": CREATE
Source changed "C:\\Hugo\\example.com\\content\\en\\cities\\204\\index.md": CREATE
Source changed "C:\\Hugo\\example.com\\content\\en\\cities\\205\\index.md": CREATE
Total in 90 ms
adding created directory to watchlist C:\Hugo\example.com\content\en\cities\230
adding created directory to watchlist C:\Hugo\example.com\content\en\cities\234
adding created directory to watchlist C:\Hugo\example.com\content\en\cities\239
adding created directory to watchlist C:\Hugo\example.com\content\en\cities\271
adding created directory to watchlist C:\Hugo\example.com\content\de\countries
adding created directory to watchlist C:\Hugo\example.com\content\de\countries\1
adding created directory to watchlist C:\Hugo\example.com\content\de\countries\19
adding created directory to watchlist C:\Hugo\example.com\content\de\countries\22
adding created directory to watchlist C:\Hugo\example.com\content\de\countries\33
adding created directory to watchlist C:\Hugo\example.com\content\de\countries\4
adding created directory to watchlist C:\Hugo\example.com\content\de\countries\6
adding created directory to watchlist C:\Hugo\example.com\content\de\countries\77
adding created directory to watchlist C:\Hugo\example.com\content\de\countries\83
adding created directory to watchlist C:\Hugo\example.com\content\de\countries\9
adding created directory to watchlist C:\Hugo\example.com\content\de\countries\98
adding created directory to watchlist C:\Hugo\example.com\content\de\regions
adding created directory to watchlist C:\Hugo\example.com\content\de\regions\10
adding created directory to watchlist C:\Hugo\example.com\content\de\regions\11
adding created directory to watchlist C:\Hugo\example.com\content\de\regions\17
adding created directory to watchlist C:\Hugo\example.com\content\de\regions\2
adding created directory to watchlist C:\Hugo\example.com\content\de\regions\20
adding created directory to watchlist C:\Hugo\example.com\content\de\regions\23
adding created directory to watchlist C:\Hugo\example.com\content\de\regions\25
adding created directory to watchlist C:\Hugo\example.com\content\de\regions\5
adding created directory to watchlist C:\Hugo\example.com\content\de\regions\7

Change detected, rebuilding site.
2019-08-10 15:41:47.121 +0200
Source changed "C:\\Hugo\\example.com\\content\\en\\cities\\230\\index.md": CREATE
Source changed "C:\\Hugo\\example.com\\content\\en\\cities\\234\\index.md": CREATE
Source changed "C:\\Hugo\\example.com\\content\\en\\cities\\239\\index.md": CREATE
Source changed "C:\\Hugo\\example.com\\content\\en\\cities\\271\\index.md": CREATE
Source changed "C:\\Hugo\\example.com\\content\\de\\countries\\1\\_index.md": CREATE
Source changed "C:\\Hugo\\example.com\\content\\de\\countries\\19\\_index.md": CREATE
Source changed "C:\\Hugo\\example.com\\content\\de\\countries\\22\\_index.md": CREATE
[signal 0xc0000005 code=0x0 addr=0x48 pc=0x132c647]

goroutine 290 [running]:
github.com/gohugoio/hugo/commands.(*staticSyncer).isStatic(...)
        /root/project/hugo/commands/static_syncer.go:36
github.com/gohugoio/hugo/commands.(*commandeer).handleEvents.func2(0xc001bcfec0, 0x37, 0x1ce5e00, 0xc001c78e20, 0x0, 0x0, 0xc001ac1738, 0x58b3b5)
        /root/project/hugo/commands/hugo.go:1029 +0x1b7
github.com/gohugoio/hugo/hugofs.(*Walkway).walk(0xc000495710, 0xc001bcfec0, 0x37, 0x1ce5e00, 0xc001c78e20, 0x0, 0x0, 0x0, 0xc0022e7860, 0x0, ...)
        /root/project/hugo/hugofs/walk.go:167 +0x81
github.com/gohugoio/hugo/hugofs.(*Walkway).walk(0xc000495710, 0xc0014186c0, 0x2d, 0x1ce5e00, 0xc001ccb180, 0xc00230d0c0, 0x1, 0x1, 0xc0022e7860, 0x1cb6560, ...)
        /root/project/hugo/hugofs/walk.go:290 +0x777
github.com/gohugoio/hugo/hugofs.(*Walkway).Walk(0xc000495710, 0xc001ccaf60, 0xc0014186c0)
        /root/project/hugo/hugofs/walk.go:137 +0x1e4
github.com/gohugoio/hugo/helpers.SymbolicWalk(0x1cf2680, 0x293a280, 0xc0014186c0, 0x2d, 0xc0022e7860, 0x0, 0x0)
        /root/project/hugo/helpers/path.go:518 +0x11b
github.com/gohugoio/hugo/commands.(*commandeer).handleEvents(0xc000076680, 0xc0018b8e60, 0xc001c6e908, 0xc002258000, 0x26, 0x40, 0xc001ff8000)
        /root/project/hugo/commands/hugo.go:1042 +0x1277
github.com/gohugoio/hugo/commands.(*commandeer).newWatcher.func1(0xc0018b8e60, 0xc000076680, 0xc001c6e908, 0xc001ff8000)
        /root/project/hugo/commands/hugo.go:842 +0x1df
created by github.com/gohugoio/hugo/commands.(*commandeer).newWatcher
        /root/project/hugo/commands/hugo.go:838 +0x295

Schwarz@TSchwarz MINGW64 /c/Hugo/example.com (master)
$ hugo version
Hugo Static Site Generator v0.56.3/extended windows/amd64 BuildDate: unknown
```

<img width="698" alt="스크린샷 2019-08-14 오후 11 33 06" src="https://user-images.githubusercontent.com/4777069/63029842-3ad35b00-beec-11e9-9d13-6ac08389aaa2.png">
Hugo Static Site Generator v0.56.3/extended darwin/amd64 BuildDate: unknown

## reappearance
```
$ hugo new site sample
$ cd sample
$ touch create.sh parallel.sh
$ chmod +x create.sh parallel.sh
```
`create.sh`:
```
#!/bin/bash

for (( i = $1 ; i <= $2 ; i++ )); do
    hugo new post/"${i}"/index.md
done
```
`parallel.sh`:
```
#!/bin/bash

$1 1 1000 &
$1 1001 2000 &
$1 2001 3000 &
```

and open 3 terminals.

- first : run `hugo serve -D`
- second : run `./parallel.sh ./create.sh`. the following message is printed: '.../post/`$i`/index.md'. 
- third : every time $ i changes between 100 and 200, run the following command: `rm -rf content/post/*`

since this issue is multithreaded, you may need to repeat the command multiple times in the third terminal.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x60 pc=0x4efbc40]

.....
```

If the reproduction is successful, you will see the above message. the above appears to be a problem in `/root/project/hugo/commands/static_syncer.go:36`, but in fact more errors can occur.

## cause

first, I checked `isStatic()` in the `static_syncer.go` file.

```
func (s *staticSyncer) isStatic(filename string) bool {
	return s.c.hugo.BaseFs.SourceFilesystems.IsStatic(filename)
}
```

as the error suggests, the wrong memory reference is the problem. To see this, I added a simple code above `return`.

```
if s == nil {
	fmt.Println("sync is nil")
} else if s.c == nil {
	fmt.Println("commandeer is nil")
} else if s.c.hugo == nil {
	fmt.Println("hugo is nil")
} else if s.c.hugo.BaseFs == nil {
	fmt.Println("basefs is nil")
} else if s.c.hugo.BaseFs.SourceFilesystems == nil {
	fmt.Println("source file system is nil")
}
```

The result is `hugo is nil`. Follow the code a little bit and you will see that the field is in [`commandeerHugoState`](https://github.com/gohugoio/hugo/blob/4644b95bd568946429482aa36eeaff1eec6a7075/commands/commandeer.go#L55).

If this field is nil:
1. `commandeer` changed: In this case, `commandeer is nil` would have been output in the if statement.
2. `commandeerHugoState` changed: Assigned in two places. [`commandeer.go`](https://github.com/gohugoio/hugo/blob/4644b95bd568946429482aa36eeaff1eec6a7075/commands/commandeer.go#L157), [`hugo.go`](https://github.com/gohugoio/hugo/blob/4644b95bd568946429482aa36eeaff1eec6a7075/commands/hugo.go#L778)
3. `commandeerHugoState.hugo` changed: Assigned from [`commandeer.go`](https://github.com/gohugoio/hugo/blob/4644b95bd568946429482aa36eeaff1eec6a7075/commands/commandeer.go#L382)

I checked if nil is assigned in `commandeerHugoState.hugo`, but the sample didn't have that problem.

If you look at the code a little bit further, you can see that all three of these things run on different goroutines.
- reassign `commandeerHugoState`
- [initialize `commandeerHugoState.hugo`](https://github.com/gohugoio/hugo/blob/4644b95bd568946429482aa36eeaff1eec6a7075/commands/commandeer.go#L341-L384)
- access `commandeerHugoState.hugo`

At this point, `commandeerHugoState.hugo` initialization always runs after `commandeerHugoState` reallocation. however, `commandeerHugoState.hugo` access is independent of other goroutines.

In this case, this problem occurs when each goroutine is executed in the following order.
1. reassign `commandeerHugoState`
2. __access `commandeerHugoState.hugo` (is nil, panic)__
3. initialize `commandeerHugoState.hugo`

--- a/commands/commandeer.go
+++ b/commands/commandeer.go
@@ -52,8 +52,9 @@ import (
 
 type commandeerHugoState struct {
 	*deps.DepsCfg
-	hugo     *hugolib.HugoSites
-	fsCreate sync.Once
+	hugoSites *hugolib.HugoSites
+	fsCreate  sync.Once
+	created   chan struct{}
 }
 
 type commandeer struct {
@@ -95,6 +96,17 @@ type commandeer struct {
 
 	// Any error from the last build.
 	buildErr error
+}
+
+func newCommandeerHugoState() *commandeerHugoState {
+	return &commandeerHugoState{
+		created: make(chan struct{}),
+	}
+}
+
+func (c *commandeerHugoState) hugo() *hugolib.HugoSites {
+	<-c.created
+	return c.hugoSites
 }
 
 func (c *commandeer) errCount() int {
@@ -154,7 +166,7 @@ func newCommandeer(mustHaveConfigFile, running bool, h *hugoBuilderCommon, f fla
 	c := &commandeer{
 		h:                   h,
 		ftch:                f,
-		commandeerHugoState: &commandeerHugoState{},
+		commandeerHugoState: newCommandeerHugoState(),
 		doWithCommandeer:    doWithCommandeer,
 		visitedURLs:         types.NewEvictingStringQueue(10),
 		debounce:            rebuildDebouncer,
@@ -373,13 +385,15 @@ func (c *commandeer) loadConfig(mustHaveConfigFile, running bool) error {
 
 		err = c.initFs(fs)
 		if err != nil {
+			close(c.created)
 			return
 		}
 
 		var h *hugolib.HugoSites
 
 		h, err = hugolib.NewHugoSites(*c.DepsCfg)
-		c.hugo = h
+		c.hugoSites = h
+		close(c.created)
 
 	})
 

--- a/commands/deploy.go
+++ b/commands/deploy.go
@@ -58,7 +58,7 @@ documentation.
 			if err != nil {
 				return err
 			}
-			deployer, err := deploy.New(comm.Cfg, comm.hugo.PathSpec.PublishFs)
+			deployer, err := deploy.New(comm.Cfg, comm.hugo().PathSpec.PublishFs)
 			if err != nil {
 				return err
 			}

--- a/commands/hugo.go
+++ b/commands/hugo.go
@@ -92,7 +92,7 @@ func Execute(args []string) Response {
 
 	if c == cmd && hugoCmd.c != nil {
 		// Root command executed
-		resp.Result = hugoCmd.c.hugo
+		resp.Result = hugoCmd.c.hugo()
 	}
 
 	if err == nil {
@@ -338,16 +338,16 @@ func (c *commandeer) fullBuild() error {
 		}
 	}
 
-	for _, s := range c.hugo.Sites {
+	for _, s := range c.hugo().Sites {
 		s.ProcessingStats.Static = langCount[s.Language().Lang]
 	}
 
 	if c.h.gc {
-		count, err := c.hugo.GC()
+		count, err := c.hugo().GC()
 		if err != nil {
 			return err
 		}
-		for _, s := range c.hugo.Sites {
+		for _, s := range c.hugo().Sites {
 			// We have no way of knowing what site the garbage belonged to.
 			s.ProcessingStats.Cleaned = uint64(count)
 		}
@@ -483,7 +483,7 @@ func (c *commandeer) build() error {
 	// TODO(bep) Feedback?
 	if !c.h.quiet {
 		fmt.Println()
-		c.hugo.PrintProcessingStats(os.Stdout)
+		c.hugo().PrintProcessingStats(os.Stdout)
 		fmt.Println()
 
 		if createCounter, ok := c.destinationFs.(hugofs.DuplicatesReporter); ok {
@@ -539,7 +539,7 @@ func (c *commandeer) serverBuild() error {
 	// TODO(bep) Feedback?
 	if !c.h.quiet {
 		fmt.Println()
-		c.hugo.PrintProcessingStats(os.Stdout)
+		c.hugo().PrintProcessingStats(os.Stdout)
 		fmt.Println()
 	}
 
@@ -558,7 +558,7 @@ func (c *commandeer) doWithPublishDirs(f func(sourceFs *filesystems.SourceFilesy
 
 	langCount := make(map[string]uint64)
 
-	staticFilesystems := c.hugo.BaseFs.SourceFilesystems.Static
+	staticFilesystems := c.hugo().BaseFs.SourceFilesystems.Static
 
 	if len(staticFilesystems) == 0 {
 		c.logger.INFO.Println("No static directories found to sync")
@@ -610,7 +610,7 @@ func chmodFilter(dst, src os.FileInfo) bool {
 }
 
 func (c *commandeer) copyStaticTo(sourceFs *filesystems.SourceFilesystem) (uint64, error) {
-	publishDir := c.hugo.PathSpec.PublishDir
+	publishDir := c.hugo().PathSpec.PublishDir
 	// If root, remove the second '/'
 	if publishDir == "//" {
 		publishDir = helpers.FilePathSeparator
@@ -655,7 +655,7 @@ func (c *commandeer) copyStaticTo(sourceFs *filesystems.SourceFilesystem) (uint6
 }
 
 func (c *commandeer) firstPathSpec() *helpers.PathSpec {
-	return c.hugo.Sites[0].PathSpec
+	return c.hugo().Sites[0].PathSpec
 }
 
 func (c *commandeer) timeTrack(start time.Time, name string) {
@@ -689,7 +689,7 @@ func (c *commandeer) getDirList() ([]string, error) {
 
 	}
 
-	watchDirs := c.hugo.PathSpec.BaseFs.WatchDirs()
+	watchDirs := c.hugo().PathSpec.BaseFs.WatchDirs()
 	for _, watchDir := range watchDirs {
 
 		w := hugofs.NewWalkway(hugofs.WalkwayConfig{Logger: c.logger, Info: watchDir, WalkFn: walkFn})
@@ -704,7 +704,7 @@ func (c *commandeer) getDirList() ([]string, error) {
 }
 
 func (c *commandeer) buildSites() (err error) {
-	return c.hugo.Build(hugolib.BuildCfg{})
+	return c.hugo().Build(hugolib.BuildCfg{})
 }
 
 func (c *commandeer) handleBuildErr(err error, msg string) {
@@ -726,16 +726,16 @@ func (c *commandeer) rebuildSites(events []fsnotify.Event) error {
 
 		// Make sure we always render the home pages
 		for _, l := range c.languages {
-			langPath := c.hugo.PathSpec.GetLangSubDir(l.Lang)
+			langPath := c.hugo().PathSpec.GetLangSubDir(l.Lang)
 			if langPath != "" {
 				langPath = langPath + "/"
 			}
-			home := c.hugo.PathSpec.PrependBasePath("/"+langPath, false)
+			home := c.hugo().PathSpec.PrependBasePath("/"+langPath, false)
 			visited[home] = true
 		}
 
 	}
-	return c.hugo.Build(hugolib.BuildCfg{RecentlyVisited: visited}, events...)
+	return c.hugo().Build(hugolib.BuildCfg{RecentlyVisited: visited}, events...)
 }
 
 func (c *commandeer) partialReRender(urls ...string) error {
@@ -744,7 +744,7 @@ func (c *commandeer) partialReRender(urls ...string) error {
 	for _, url := range urls {
 		visited[url] = true
 	}
-	return c.hugo.Build(hugolib.BuildCfg{RecentlyVisited: visited, PartialReRender: true})
+	return c.hugo().Build(hugolib.BuildCfg{RecentlyVisited: visited, PartialReRender: true})
 }
 
 func (c *commandeer) fullRebuild(changeType string) {
@@ -775,7 +775,7 @@ func (c *commandeer) fullRebuild(changeType string) {
 
 		defer c.timeTrack(time.Now(), "Total")
 
-		c.commandeerHugoState = &commandeerHugoState{}
+		c.commandeerHugoState = newCommandeerHugoState()
 		err := c.loadConfig(true, true)
 		if err != nil {
 			// Set the processing on pause until the state is recovered.
@@ -951,7 +951,7 @@ func (c *commandeer) handleEvents(watcher *watcher.Batcher,
 	filtered := []fsnotify.Event{}
 	for _, ev := range evs {
 		// Check the most specific first, i.e. files.
-		contentMapped := c.hugo.ContentChanges.GetSymbolicLinkMappings(ev.Name)
+		contentMapped := c.hugo().ContentChanges.GetSymbolicLinkMappings(ev.Name)
 		if len(contentMapped) > 0 {
 			for _, mapped := range contentMapped {
 				filtered = append(filtered, fsnotify.Event{Name: mapped, Op: ev.Op})
@@ -963,7 +963,7 @@ func (c *commandeer) handleEvents(watcher *watcher.Batcher,
 
 		dir, name := filepath.Split(ev.Name)
 
-		contentMapped = c.hugo.ContentChanges.GetSymbolicLinkMappings(dir)
+		contentMapped = c.hugo().ContentChanges.GetSymbolicLinkMappings(dir)
 
 		if len(contentMapped) == 0 {
 			filtered = append(filtered, ev)
@@ -997,7 +997,7 @@ func (c *commandeer) handleEvents(watcher *watcher.Batcher,
 		if istemp {
 			continue
 		}
-		if c.hugo.Deps.SourceSpec.IgnoreFile(ev.Name) {
+		if c.hugo().Deps.SourceSpec.IgnoreFile(ev.Name) {
 			continue
 		}
 		// Sometimes during rm -rf operations a '"": REMOVE' is triggered. Just ignore these
@@ -1073,7 +1073,7 @@ func (c *commandeer) handleEvents(watcher *watcher.Batcher,
 			// force refresh when more than one file
 			if len(staticEvents) == 1 {
 				ev := staticEvents[0]
-				path := c.hugo.BaseFs.SourceFilesystems.MakeStaticPathRelative(ev.Name)
+				path := c.hugo().BaseFs.SourceFilesystems.MakeStaticPathRelative(ev.Name)
 				path = c.firstPathSpec().RelURL(helpers.ToSlashTrimLeading(path), false)
 				livereload.RefreshPath(path)
 			} else {
@@ -1119,7 +1119,7 @@ func (c *commandeer) handleEvents(watcher *watcher.Batcher,
 
 				if navigate {
 					if onePageName != "" {
-						p = c.hugo.GetContentPage(onePageName)
+						p = c.hugo().GetContentPage(onePageName)
 					}
 				}
 

--- a/commands/mod.go
+++ b/commands/mod.go
@@ -157,7 +157,7 @@ Also note that if you configure a positive maxAge for the "modules" file cache, 
 					return err
 				}
 
-				_, err = com.hugo.FileCaches.ModulesCache().Prune(true)
+				_, err = com.hugo().FileCaches.ModulesCache().Prune(true)
 				return err
 
 			},
@@ -176,7 +176,7 @@ func (c *modCmd) withModsClient(failOnMissingConfig bool, f func(*modules.Client
 		return err
 	}
 
-	return f(com.hugo.ModulesClient)
+	return f(com.hugo().ModulesClient)
 }
 
 func (c *modCmd) initConfig(failOnNoConfig bool) (*commandeer, error) {

--- a/commands/new.go
+++ b/commands/new.go
@@ -85,13 +85,13 @@ func (n *newCmd) newContent(cmd *cobra.Command, args []string) error {
 
 	var kind string
 
-	createPath, kind = newContentPathSection(c.hugo, createPath)
+	createPath, kind = newContentPathSection(c.hugo(), createPath)
 
 	if n.contentType != "" {
 		kind = n.contentType
 	}
 
-	return create.NewContent(c.hugo, kind, createPath)
+	return create.NewContent(c.hugo(), kind, createPath)
 }
 
 func mkdir(x ...string) {

--- a/commands/new_theme.go
+++ b/commands/new_theme.go
@@ -63,7 +63,7 @@ func (n *newThemeCmd) newTheme(cmd *cobra.Command, args []string) error {
 		return newUserError("theme name needs to be provided")
 	}
 
-	createpath := c.hugo.PathSpec.AbsPathify(filepath.Join(c.Cfg.GetString("themesDir"), args[0]))
+	createpath := c.hugo().PathSpec.AbsPathify(filepath.Join(c.Cfg.GetString("themesDir"), args[0]))
 	jww.FEEDBACK.Println("Creating theme at", createpath)
 
 	cfg := c.DepsCfg

--- a/commands/server.go
+++ b/commands/server.go
@@ -244,7 +244,7 @@ func (sc *serverCmd) server(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	for _, s := range c.hugo.Sites {
+	for _, s := range c.hugo().Sites {
 		s.RegisterMediaTypes()
 	}
 
@@ -303,9 +303,9 @@ func (f *fileServer) createEndpoint(i int) (*http.ServeMux, string, string, erro
 		publishDir = filepath.Join(publishDir, root)
 	}
 
-	absPublishDir := f.c.hugo.PathSpec.AbsPathify(publishDir)
+	absPublishDir := f.c.hugo().PathSpec.AbsPathify(publishDir)
 
-	jww.FEEDBACK.Printf("Environment: %q", f.c.hugo.Deps.Site.Hugo().Environment)
+	jww.FEEDBACK.Printf("Environment: %q", f.c.hugo().Deps.Site.Hugo().Environment)
 
 	if i == 0 {
 		if f.s.renderToDisk {
@@ -398,7 +398,7 @@ func removeErrorPrefixFromLog(content string) string {
 }
 func (c *commandeer) serve(s *serverCmd) error {
 
-	isMultiHost := c.hugo.IsMultihost()
+	isMultiHost := c.hugo().IsMultihost()
 
 	var (
 		baseURLs []string
@@ -406,17 +406,17 @@ func (c *commandeer) serve(s *serverCmd) error {
 	)
 
 	if isMultiHost {
-		for _, s := range c.hugo.Sites {
+		for _, s := range c.hugo().Sites {
 			baseURLs = append(baseURLs, s.BaseURL.String())
 			roots = append(roots, s.Language().Lang)
 		}
 	} else {
-		s := c.hugo.Sites[0]
+		s := c.hugo().Sites[0]
 		baseURLs = []string{s.BaseURL.String()}
 		roots = []string{""}
 	}
 
-	templ, err := c.hugo.TextTmpl.Parse("__default_server_error", buildErrorTemplate)
+	templ, err := c.hugo().TextTmpl.Parse("__default_server_error", buildErrorTemplate)
 	if err != nil {
 		return err
 	}

--- a/commands/static_syncer.go
+++ b/commands/static_syncer.go
@@ -33,14 +33,14 @@ func newStaticSyncer(c *commandeer) (*staticSyncer, error) {
 }
 
 func (s *staticSyncer) isStatic(filename string) bool {
-	return s.c.hugo.BaseFs.SourceFilesystems.IsStatic(filename)
+	return s.c.hugo().BaseFs.SourceFilesystems.IsStatic(filename)
 }
 
 func (s *staticSyncer) syncsStaticEvents(staticEvents []fsnotify.Event) error {
 	c := s.c
 
 	syncFn := func(sourceFs *filesystems.SourceFilesystem) (uint64, error) {
-		publishDir := c.hugo.PathSpec.PublishDir
+		publishDir := c.hugo().PathSpec.PublishDir
 		// If root, remove the second '/'
 		if publishDir == "//" {
 			publishDir = helpers.FilePathSeparator

